test/Test/Database/LSMTree/Normal/StateMachine.hs

```diff
@@ -1461,6 +1469,18 @@ updateStats action lookUp result =
                                                     (numActionsPerTable stats)
               }
 
+    updClosedTableSizes stats = case action of
+        Close tableVar
+          | MTableHandle th <- lookUp tableVar
+          , let tid          = Model.tableHandleID th
+            -- This lookup can fail if the table was already closed:
+          , Just (_, table) <- Map.lookup tid (Model.tableHandles modelBefore)
+          , let  tsize       = Model.withSomeTable Model.size table
+          -> stats {
+               closedTableSizes = Map.insert tid tsize (closedTableSizes stats)
+             }
+        _ -> stats
```

Comment:
Yes. I suppose the other way to do it would be look for a successful result, in which case the Map.lookup should always succeed. Perhaps that's the clearer way round to do it.

Reply:
Actually, that doesn't work. The way I did it originally works.